cybozu/tcp.hpp

```diff
@@ -190,6 +190,12 @@ class tcp_socket: public resource {
         m_cond_write.notify_all();
     }
 
+    // This method called the invocation of <send>, <sendv>, <send_close>
```

Comment:
```
This method will be called everytime when <send>, <sendv>, 
<send_close>, or <sendv_close> is blocked because of internal buffer full.

Subclasses can override this to handle the buffer full event.
```